components/org.wso2.carbon.identity.media/src/main/java/org/wso2/carbon/identity/media/file/FileBasedStorageSystemImpl.java

```diff
@@ -487,4 +409,69 @@ private void storeFileSecurityMetadata(FileSecurity fileSecurity, JSONObject met
         metadata.put(METADATA_FILE_SECURITY, fileSecurityJSON);
     }
 
+    private void deleteMediaFile(String id, String type, String tenantDomain) throws IOException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+
+        Path fileStorageLocation = getStorageDirectory(type, tenantId, id);
+        if (fileStorageLocation != null) {
+            Path filePath = fileStorageLocation.resolve(id).normalize();
+            String metadataFileName = id + METADATA_FILE_SUFFIX + METADATA_FILE_EXTENSION;
+            Path metadataFilePath = fileStorageLocation.resolve(metadataFileName).normalize();
+            Files.deleteIfExists(filePath);
+            Files.deleteIfExists(metadataFilePath);
+        }
+    }
+
+    private Path getStorageDirectory(String fileType, int tenantId, String id) {
+
+        Path fileStorageLocation = getFileStorageLocation(fileType);
+        Path mediaPath = null;
+        if (fileStorageLocation != null) {
+            if (Files.notExists(fileStorageLocation)) {
+                return null;
+            }
+            mediaPath = fileStorageLocation.resolve(String.valueOf(tenantId));
+            if (Files.notExists(mediaPath)) {
+                return null;
+            }
+        }
+        return getUniqueDirectoryStructure(mediaPath, id);
+    }
+
+    private Path getFileStorageLocation(String fileType) {
+
+        Path fileStorageLocation = null;
+        Path configurableRootFolder = null;
+        String systemPropertyForRootFolder = System.getProperty(CONFIGURABLE_UPLOAD_LOCATION);
+        if (systemPropertyForRootFolder != null) {
+            configurableRootFolder = Paths.get(systemPropertyForRootFolder);
+        }
+
+        if (configurableRootFolder == null) {
+            configurableRootFolder = Paths.get(System.getProperty(SYSTEM_PROPERTY_CARBON_HOME));
+        }
+
+        if (configurableRootFolder != null) {
+            fileStorageLocation = configurableRootFolder.resolve(Paths.get(MEDIA_STORE + fileType));
+        }
+        return fileStorageLocation;
+    }
+
+    private Path getUniqueDirectoryStructure(Path path, String uuid) {
+
+        String[] uuidSplit = uuid.split("-");
```

Comment:
can uuid be null? if so handle null check

Reply:
uuid should always be present in the requested URL as a path param to trigger this method. therefore null check is not required.